### PR TITLE
remove sourcing check on procurement validation

### PR DESCRIPTION
**Pull request by lepistone**
Supersedes #106.

This had two problems, and an attempt to fix did not work:
1. For MTO orders, this error in produced in background in the scheduler,
   and the error is not shown to the user. An exception blocks the
   scheduler completely in that case, and also breaks all subsequent
   runs.
2. With transit locations, the check fails because it compared "Customer"
   and "Transit". This situation is actually OK.

In an attempt to fix 1, I tried to raise the exception only when in
foreground and log an exception to the procurement otherwise. This
works, but in the MTO case the user does not see the problem and the
order is validated (so it is too late to change it).

In an attempt to fix 2, I changed the check to use the "final
destination" of the first move that generated the procurement. That
fixed the transit case, but broke the MTO case, where a BUY procurement
in Stock is chained to a MTO procurement in Customers.

There are already other consistency checks: the route of the sale order
line is changed automatically when a PO is chosen, and a sale exception
does a consistency check as well.

TL;DR: I am removing this check.

### CODE REVIEW

**Comment by yvaucher** [comment]
Shouldn't this check be removed only if we add a key like 'automated' in context?
Isn't this warning useful when manually sourcing?

**Comment by lepistone** [comment]
@yvaucher I tried! but it doesn not make sense. See the write-up above.

**Comment by jgrandguillaume** [comment]
A pity, but I agree we have no choice but to remove it. So :+1:

**Comment by yvaucher** [comment]
Ok the scheduler shouldn't be blocked, so :+1:
